src/database/index.tsx

```diff
@@ -482,3 +486,75 @@ export const deleteFacility = async (
     console.error(error);
   }
 };
+
+export const duplicateScenario = async (
+  scenarioId: string,
+): Promise<Scenario | void> => {
+  try {
+    const scenario = await getScenario(scenarioId);
+
+    if (!scenario) {
+      console.error(`No scenario found for scenario: ${scenarioId}`);
+      return;
+    }
+
+    const userId = currrentUserId();
+    const timestamp = currrentTimestamp();
+    const db = await getDb();
+    const batch = db.batch();
+
+    // Duplicate and save the Scenario
+    const scenarioDoc = db.collection(scenariosCollectionId).doc();
+
+    batch.set(scenarioDoc, {
+      name: `Copy of ${scenario.name}`,
+      description: `This is a copy of the '${
+        scenario.name
+      }' scenario, made on ${format(new Date(), MMMMdyyyy)}`,
+      baseline: false,
+      dailyReports: false,
+      dataSharing: false,
+      promoStatuses: {},
+      roles: {
+        [userId]: "owner",
+      },
+      createdAt: timestamp,
+      updatedAt: timestamp,
+    });
+
+    // Duplicate and save all of the Facilities
+    const facilities = await getFacilities(scenarioId);
+    for (const facility of facilities) {
+      const facilityCopy = Object.assign({}, facility);
+      delete facilityCopy.id;
+      delete facilityCopy.scenarioId;
```

Comment:
Note that I am intentionally deleting the `facilityCopy`'s id so that the copied over facilities will get their own unique id.  However, I am intentionally leaving their original `createdAt` and `updatedAt` timestamps.